File: Tarea 1/ejercicio3.py
import random

sucursales = [f"Sucursal {i+1}" for i in range(25)]
ventas = [random.randint(10000, 50000) for _ in range(25)]

prom = sum(ventas) / len(ventas)

sucursalessp = []
for i in range(25):
    if ventas[i] > prom:
        sucursalessp.append((sucursales[i], ventas[i]))

print(f"Promedio de ventas: {prom:.2f}")
print("\nSucursales con ventas mayores al promedio:")
for sucursal, venta in sucursalessp:
    print(f"{sucursal}: {venta}") 

File: ejercicio3.cpp
#include <iostream>
#include <vector>
#include <string>
#include <ctime>       
#include <chrono>      

using namespace std;
using namespace chrono;

int main() {
    auto inicio = high_resolution_clock::now();

    vector<string> sucursales;
    vector<int> ventas;

    srand(time(0));

    for (int i = 0; i < 25; ++i) {
        sucursales.push_back("Sucursal " + to_string(i + 1));
        ventas.push_back(rand() % (50000 - 10000 + 1) + 10000);  
    }
    int suma = 0;
    for (int venta : ventas) {
        suma += venta;
    }
    double promedio = static_cast<double>(suma) / ventas.size();

    cout << "Promedio de ventas: " << promedio << endl;

    cout << "\nSucursales con ventas mayores al promedio:\n";
    for (int i = 0; i < 25; ++i) {
        if (ventas[i] > promedio) {
            cout << sucursales[i] << ": " << ventas[i] << endl;
        }
    }

    auto fin = high_resolution_clock::now();
    auto duracion = duration_cast<milliseconds>(fin - inicio);

    cout << "\nTiempo de ejecucion: " << duracion.count() << " ms" << endl;

    return 0;
}
